Install cmdstan via r

jgabry opened the pull request:
#### Submission Checklist

- [x] Run unit tests
- [x] Declare copyright holder and agree to license (see below)

#### Summary

Closes #88 

(@rok-cesnovar I know I've opened a lot of PRs recently, so no worries if it takes you some time to get to this!)

This PR changes `install_cmdstan()` so that it installs CmdStan without needing the script `make_cmdstan.sh`. Standard R functions are used for downloading and unzipping the tarball before running make via `processx::run()`. 


I also updated the travis and appveyor scripts to use `install_cmdstan()` instead of `make_cmdstan.sh`. The only reason I haven't removed `make_cmdstan.sh` entirely is that it's still used by `cmdstan_git_checkout_branch()`, which I haven't touched in this PR (although we could do something similar for that).  

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting 
(this will be you or your assignee, such as a university or company): 
**Columbia University**

By submitting this pull request, the copyright holder is agreeing to 
license the submitted work under the following licenses:

- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)

rok-cesnovar commented on tests/testthat/test-install.R:
```diff
@@ -1,35 +1,51 @@
-      install_cmdstan(dir = dir, cores = 1, quiet = FALSE, overwrite = FALSE),
-      "Please remove or rename the 'cmdstan' folder"
+    expect_warning(
+      install_cmdstan(dir = dir, overwrite = FALSE),
```
I think there is a quiet = FALSE missing here? This tests is probably skipped on Travis and that is why its not failing?

jgabry replied:
It was actually a different reason, but fixed now!

rok-cesnovar commented on R/install.R:
```diff
@@ -128,3 +152,82 @@ cmdstan_git_checkout_branch <- function(repo_branch,
+    if (!quiet) {
+      suggestion <- "See the error message(s) above."
+    } else {
+      suggestion <- "Please try again with 'quiet=FALSE' to get the full error output."
```
The error messages will be output here in any case because of the 

    stderr_line_callback = function(x,p) { if(quiet) message(x) }

So maybe just write out "suggestion <- "See the error message(s) above or try again with 'quiet=FALSE' to get the full installation output."

jgabry commented on R/install.R:
```diff
@@ -128,3 +152,82 @@ cmdstan_git_checkout_branch <- function(repo_branch,
+    spinner = quiet,
+    error_on_status = FALSE,
+    stderr_line_callback = function(x,p) { if(quiet) message(x) },
+    timeout = 600 # timeout after 10 minutes to avoid infinite loop problem?
```
Good call. I’ll make it an argument to install_cmdstan.

jgabry commented:
Thanks for the review! Will make those changes shortly

jgabry commented:
@rok-cesnovar Waiting to see if travis and appveyor pass now, but I addressed all the comments so I think this is ready for review again.  

> Should we alse delete the bash script in this PR?

The only reason we can't delete the bash script is because it's still used by the `cmdstan_git_checkout_branch()` function. We can definitely change `cmdstan_git_checkout_branch()` to not use it either, but maybe as a separate PR?

jgabry commented:
Ok great thanks. I’ll merge if CI passes.

jgabry commented:
Looks like Travis is passing but timing out during codecov. I’ll try adding travis_wait as described here:

https://docs.travis-ci.com/user/common-build-problems/#build-times-out-because-no-output-was-received
